ext/DifferentiationInterfaceEnzymeExt/reverse.jl

```diff
@@ -1,6 +1,15 @@
 const AutoReverseEnzyme = AutoEnzyme{Val{:reverse}}
 DI.autodiff_mode(::AutoReverseEnzyme) = DI.ReverseMode()
-DI.handles_output_type(::AutoReverseEnzyme, ::Type{<:AbstractArray}) = false
+
+struct Mutating{F}
+    f::F
+end
+
+function (m::Mutating{F})(y::AbstractArray, x) where {F}
```

Comment:
Why `where {F}`?

Reply:
to force specialization

Reply:
https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing

Reply:
not sure it's useful here tho